### PR TITLE
chore: [jsonwebtoken] remove undefined in SignOptions.expiresIn type

### PR DESCRIPTION
**Pull request by Mifrill**
📓 After release [8.0.0](https://github.com/auth0/node-jsonwebtoken/compare/v7.4.3...v8.0.0), `SignOptions.expiresIn` throw an error when given `undefined`.

```ts
expiresIn: { isValid: function(value) { return isInteger(value) || isString(value); }, message: '"expiresIn" should be a number of seconds or string representing a timespan' },
```
🗒️ Source: https://github.com/auth0/node-jsonwebtoken/blob/f3138506fe7fb9383a709e97c12e96148f0c7d7b/sign.js#L14

📝 https://github.com/DefinitelyTyped/DefinitelyTyped/pull/54322#discussion_r1480596340

- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `pnpm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the `package.json`.

### CODE REVIEW

**Review by sandersn — APPROVED** [review]

**Comment by sandersn** [comment]
I'm not sure how the bot handles approved drafts, so ping me if it doesn't offer to merge it after you mark it Ready for Review.

**Comment by DangerBotOSS** [comment]
## Formatting

The following files are not formatted:
1. types/jsonwebtoken/jsonwebtoken-tests.ts

Consider running `pnpm dprint fmt` on these files to make review easier.
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 78781a3bf18a762e5abbb92f3adbe5c5990166a5
</p>

File: types/jsonwebtoken/tsconfig.json
Comment (Mifrill) [inline types/jsonwebtoken/tsconfig.json:14]: 📝 https://www.typescriptlang.org/tsconfig#exactOptionalPropertyTypes

**Comment by Mifrill** [comment]
Hey @sandersn thanks for the review. Could you recommend an update for [package.json version](https://github.com/Mifrill/DefinitelyTyped/blob/129645c13623d5b5ab18b5a74649e72d8eb67863/types/jsonwebtoken/package.json#L4) in this particular case?

As I see in [documentation](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#how-do-definitely-typed-package-versions-relate-to-versions-of-the-corresponding-library) that `@types packages always type packages of the same version`. So it seems like the update of new version is not needed, since package currently on version 9.0.2. On the other hand, it probably makes sense to apply those changes in older versions of a library's type declarations, because that change is affected after release [8.0.0](https://github.com/auth0/node-jsonwebtoken/compare/v7.4.3...v8.0.0).

File: types/jsonwebtoken/tsconfig.json
Comment (peterblazejewicz) [reply types/jsonwebtoken/tsconfig.json:14]: hey, those are for clients consuming definitions, in essence: typescript projects.
It's not default option, this repo (DT) provides only compatibility with this TS configuration (better worded: promotes, we advice to use it)

**Comment by peterblazejewicz** [comment]
@Mifrill 
Not sure what's the problem. The `expiresIn` is optional by design (ref).
this is from v9:
```js
jsonwebtoken.sign({foo: 123}, '123');
'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJmb28iOjEyMywiaWF0IjoxNzA3NTEwNzM4fQ.ad_FZZ27JJotbOjpCB-S3rl-EajxZUWlwKsouHYG78g'
```
https://github.com/auth0/node-jsonwebtoken/issues/795

**Comment by Mifrill** [comment]
@peterblazejewicz the problem is `expiresIn` cannot be `undefined` when specified.

See this test:
https://github.com/Mifrill/DefinitelyTyped/blob/129645c13623d5b5ab18b5a74649e72d8eb67863/types/jsonwebtoken/jsonwebtoken-tests.ts#L57-L58

The optional `expiresIn` property, can be only a string or number, but cannot be undefined. This PR is doing that strictly.

File: types/jsonwebtoken/tsconfig.json
Comment (Mifrill) [reply types/jsonwebtoken/tsconfig.json:14]: Hey @peterblazejewicz thanks for the clarification. 
This configuration option needs to ensure correct behavior for type.
I don't see any constraints in the [documentation](https://github.com/DefinitelyTyped/DefinitelyTyped#tsconfigjson) regarding this configuration option change.

**Comment by sandersn** [comment]
We don't have a policy about exactOptionalPropertyTypes in tsconfig. Given that it's an error for `expiresIn` to be undefined, I think it's OK to set exactOptionalPropertyTypes here. However, most dependents of these types aren't going to have it set, so it's really only benefiting those who do.

If this comes up more often, we should decide what to do to encourage people who install `@types/jsonwebtoken` to turn on exactOptionalPropertyTypes. However, it's been pretty rare till now to forbid `undefined` for optional properties.

**Comment by sandersn** [comment]
If you want to add types for 8.0 as well, you can add a v8/ subdirectory that has the types that were current as of version 8.5.9 and then add this fix.

**Review by jakebailey — APPROVED** [review]
I'll OK this, but you can do the v8 thing if you want (here or in another PR).

**Comment by Mifrill** [comment]
@jakebailey Thank you for the feedback 👍 I will make changes for v8 in the next PR shortly.

**Comment by Mifrill** [comment]
Ready to merge

**Pull request merged**
